contracts/mainnet/connectors/morpho-blue/main.sol

```diff
@@ -0,0 +1,918 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "./helpers.sol";
+import "./events.sol";
+
+abstract contract MorphoBlue is Helpers, Events {
+	/**
+	 * @dev Supply ETH/ERC20 Token for lending.
+	 * @notice Supplies assets to Morpho Blue for lending.
+	 * @param _marketParams The market to supply assets to. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _assets The amount of assets to supply. (For max: `uint256(-1)`)
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens deposited.
+	 */
+	function supply(
+		MarketParams memory _marketParams,
+		uint256 _assets,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt;
+		(
+			_marketParams, // Updated token contracts in case of Eth
+			_amt
+		) = _performEthToWethConversion(
+			_marketParams,
+			_assets,
+			address(this),
+			_getId,
+			Mode.Other
+		);
+
+		approve(
+			TokenInterface(_marketParams.loanToken),
+			address(MORPHO_BLUE),
+			_amt
+		);
+
+		uint256 _shares;
+		(_assets, _shares) = MORPHO_BLUE.supply(
+			_marketParams,
+			_amt,
+			0,
+			address(this),
+			new bytes(0)
+		);
+
+		setUint(_setId, _assets);
+
+		_eventName = "LogSupplyAssets((address,address,address,address,unit256),unit256,unit256,unit256,unit256)";
+		_eventParam = abi.encode(
+			_marketParams,
+			_assets,
+			_shares,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @dev Supply ETH/ERC20 Token for lending.
+	 * @notice Supplies assets to Morpho Blue for lending.
+	 * @param _marketParams The market to supply assets to. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _assets The amount of assets to supply. (For max: `uint256(-1)`)
+	 * @param _onBehalf The address that will get the shares.
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens deposited.
+	 */
+	function supplyOnBehalf(
+		MarketParams memory _marketParams,
+		uint256 _assets,
+		address _onBehalf,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt;
+		(
+			_marketParams, // Updated token contracts in case of Eth
+			_amt
+		) = _performEthToWethConversion(
+			_marketParams,
+			_assets,
+			_onBehalf,
+			_getId,
+			Mode.Other
+		);
+
+		approve(
+			TokenInterface(_marketParams.loanToken),
+			address(MORPHO_BLUE),
+			_amt
+		);
+
+		uint256 _shares;
+		(_assets, _shares) = MORPHO_BLUE.supply(
+			_marketParams,
+			_amt,
+			0,
+			_onBehalf,
+			new bytes(0)
+		);
+
+		setUint(_setId, _assets);
```

Comment:
The supplied `_assets` will increase because of interest. I think `_shares` should be the value to keep in storage, as it keeps better tracks of the position over time